core-tests/src/test/java/org/mule/runtime/core/internal/logger/CustomLogger.java

```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.logger;
+
+import static org.apache.commons.lang3.StringUtils.replaceOnce;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.Marker;
+
+/**
+ * In some tests we were replacing the original logger in a class with a mocked logger via reflection and then setting it back to
+ * the original. This reflective access will not work in Java 17, hence writing a custom logger that will be used with specific
+ * tests.
+ */
+public class CustomLogger implements Logger {
+
+  private final String name;
+  private List<String> messages;
+  private final Logger logger;
+  private boolean customLoggingEnabled;
+
+  public CustomLogger(Logger logger, String name, boolean loggingEnabled) {
+    this.name = name;
+    this.messages = new ArrayList<>();
+    this.logger = logger;
+    this.customLoggingEnabled = loggingEnabled;
```

Comment:
Some of the code we are testing has conditional logging like ExtensionMessageSource

   if (LOGGER.isInfoEnabled()) {
      LOGGER.info("Message source '{}' on flow '{}' successfully started",
                  sourceModel.getName(), getLocation().getRootContainerName());
    }
So relying on original logger to determine if logging is enabled fails ther tests. Tried addding a log4j.xml just for affected tests, but I think thats introducing lot of changes, also didnt work well.